duckduckgo/Android
# Fix download confirmation fragment illegal state exception

Task/Issue URL: https://app.asana.com/0/361428290920652/1171376595453734

**Description**:
Fixes java.lang.IllegalStateException. Only shows the dialog if the state hasn't been saved yet as per FragmentManager "operations that would change saved state should not be performed if (isStateSaved) returns true". We could show the dialog with state loss however the activity is most likely being destroyed and there is no need to now show the confirmation dialog. 

**Steps to test this PR**:
* visit https://github.com/duckduckgo/Android/releases
* download our apk
* make sure that the confirmation dialog shows (if the app doesn't have permission yet you will need to provide permission and repeat the process as we only show the dialog when other confirmation dialogs have not been displayed)

---
###### Internal references:
[Software Engineering Expectations](https://app.asana.com/0/59792373528535/199064865822552)
[Technical Design Template](https://app.asana.com/0/59792373528535/184709971311943)
